src/Resources/contao/classes/BackendTemplate.php

```diff
@@ -73,7 +73,7 @@ protected function compile()
 				$strStyleSheets .= \Template::generateStyleTag($this->addStaticUrlTo($stylesheet), $options->media);
 			}
 
-			$this->stylesheets = $strStyleSheets;
+			$this->stylesheets .= $strStyleSheets;
```

Comment:
Shouldn't `$this->stylesheets` be of type `array` instead of `string`?
```twig
% block stylesheets %}
    {% for stylesheet in stylesheets %}
        <link rel="stylesheet" href="{{ stylesheet }}">
    {% endfor %}
{% endblock %}
```

Same for `$this->javascripts` and `$this->mootols`.

Reply:
The block get's rendered to a string in the section you quoted?

I guess I don't see your point... :disappointed:
Could you clearify?

Reply:
Don't you have to pass the data to the twig template? 🤔

Reply:
I still do not know what you mean, I simply added a template, expanded the contao-core template and off I go. Pretty close to the original code by @sheeep in his custom backend controller example.
To be honest, it is almost the same, I simply wanted to also be able to add stylesheets and javascripts in addition to headline, content and error.
I'll try to clearify again.

Controller (route annotation omitted as it does not relate):
```php
class DemoController
{
    /**
     * @return array
     *
     * @Template("AcmeBundle::dummy.html.twig")
     */
    public function __invoke()
    {
        // Without this PR we are forced to do this:
        // $GLOBALS['TL_CSS'][] = 'bundles/acmebundle/css/custom';
        // $GLOBALS['TL_JAVASCRIPT'][] = 'bundles/acmebundle/js/some-script';
        // $GLOBALS['TL_MOOTOOLS'][] = '<!-- MOOO comment! -->';

        // With this PR and the enhanced twig template we can do THIS:
        return [
                'stylesheets' => ['bundles/acmebundle/css/custom'],
                'javascripts' => ['bundles/acmebundle/js/some-script'],
                'mootools' => ['<!-- MOOO comment! -->'],
        ];
    }
}
```
AcmeBundle::dummy.html.twig:
```twig
{{ render_contao_backend_template({
    main: block('main'),
    error: block('error'),
    headline: block('headline'),
    stylesheets: block('stylesheets'),
    javascripts: block('javascripts'),
    mootools: block('mootools')
}) | raw }}

{# 
  Here we generate the stylesheets to be added to be_main instead of adding 
  stuff to $GLOBALS['TL_CSS'] within a Symfony controller.
#}
{% block stylesheets %}
    {% for stylesheet in stylesheets %}
        <link rel="stylesheet" href="{{ stylesheet }}">
    {% endfor %}
{% endblock %}
{# 
  Here we generate the javascripts to be added to be_main instead of adding 
  stuff to $GLOBALS['TL_JAVASCRIPT'] within a Symfony controller.
  TODO: add support for "async" here.
#}
{% block javascripts %}
    {% for javascript in javascripts %}
        <script src="{{ javascript }}"></script>
    {% endfor %}
{% endblock %}
{# 
  Here we have the mootools scripts to be added to be_main instead of adding 
  stuff to $GLOBALS['TL_MOOTOOLS'] within a Symfony controller.
#}
{% block mootools %}
    {% for mootool in mootools %}
        {{ mootool }}
    {% endfor %}
{% endblock %}

{% block headline %}
    This is headline as in example by @sheep.
{% endblock %}

{% block main %}
Hello there - I am the main content.
{% endblock %}
```

Rendering is as follows:
1. Annotation passes returned array data to template.
2. Template prepares blocks
3. Template passes blocks to the wrapper ContaoTemplateExtension from #512
4. The ContaoTemplateExtension adds ALL blocks to template (see [here](https://github.com/contao/core-bundle/pull/512/files#diff-5fb1652178784ba302a159c90bb5c2c4R75)
5. Contao BackendTemplate class renders result.

My PR solves the issue, that the in 4. added contents of `stylesheets`, `javascripts` and `mootools` are preserved and values from `$GLOBALS` get appended instead of overwriting.

I hope now you understand what I mean.

Reply:
nevermind, I'm in the wrong context - all good 😎

Reply:
It should not be needed because `BackendTemplate` extends `Template` which extends `Controller` which extends `System` which has a `__get()` returning `null` on unknown keys.
Therefore the `getter` will bubble up to `System` and the setter will set the value in the array.

On my System appending a string to `null` did not raise any notice but maybe I forgot some flag:
```
xtra@sigyn:~$ php -ddisplay_errors=stderr -derror_reporting=-1 \
    -r '$foo = null; $foo .= "bar"; var_export($foo);'
'bar'
```
So IMO we should be good to go.

@sheeep Did you accidentally forget these values or intentionally? So, is it a bugfix or feature?

Reply:
That was not intentional. No. So this might be a bugfix instead of a feature?